exp/modeling_bibo.py

```
@@ -0,0 +1,600 @@
+            cache_position = torch.arange(
+                past_seen_tokens,
+                past_seen_tokens + seq_length,
+                device=inputs_embeds.device,
+            )
+
+        if attention_mask is not None:
+            if attention_mask.dim() != 2:
+                raise ValueError(
+                    "attention_mask must be a 2D (batch, seq) padding mask or None, "
+                    f"got {attention_mask.dim()}D"
+                )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)